File: hello-world/hello.hs.lua
--[[ Code only visible to Haskell

main = do
    putStrLn "Hello from \27[95mHaskell\27[m!"

{- Code only visible to Lua ]]
local putStrLn = print
putStrLn "Hello from \27[94mLua\27[m!"

-- Code visible to Haskell and Lua -}
    putStrLn "Hello from \27[95mHaskell\27[m and \27[94mLua\27[m!"

File: hello-world/hello_hs.jl
(#=) = True

println :: String -> IO ()
println = putStrLn

main :: IO ()
main = do
    println ("Hello from \x1b[95mHaskell\x1b[m!")

{- code only visible to Julia =#)
println("Hello from \x1b[35mJulia\x1b[m!")

# code visible to Haskell and Julia -}
    println("Hello from \x1b[95mHaskell\x1b[m and \x1b[35mJulia\x1b[m!")
